### PR TITLE
Allow cert file and private key file to be passed in

### PR DESCRIPTION
**Pull request by jkotalik**
Allows msquic to be configured on linux with openssl.

### CODE REVIEW

**Review by Tratcher — COMMENTED** [review]
Why not have people load the cert themselves and pass it in?

**Comment by jkotalik** [comment]
This is a work around for limitations in msquic. Msquic currently doesn't support supplying x509 certificates, only cert files and private key files.

I could try to reverse engineer these two files from an x509, but that seems bad. This is purely temporary for now.

**Comment by anurse** [comment]
> I could try to reverse engineer these two files from an x509, but that seems bad.

It's also going to be impossible in some circumstances (such as when using an OpenSSL "provider" for a hardware device). This seems like a reasonable compromise for the current situation.

File: src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionListener.cs [hunk 1]
```diff
@@ -29,9 +29,16 @@ public QuicConnectionListener(QuicTransportOptions options, IQuicTrace log, EndP
...
             EndPoint = endpoint;
+
+            var quicListenerOptions = new QuicListenerOptions();
             var sslConfig = new SslServerAuthenticationOptions();
             sslConfig.ServerCertificate = options.Certificate;
             sslConfig.ApplicationProtocols = new List<SslApplicationProtocol>() { new SslApplicationProtocol(options.Alpn) };
+
+            quicListenerOptions.ServerAuthenticationOptions = sslConfig;
+            quicListenerOptions.CertificateFilePath = options.CertificateFilePath;
+            quicListenerOptions.PrivateKeyFilePath = options.PrivateKeyFilePath;
```
Comment (Tratcher) [inline src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionListener.cs]: quicListenerOptions isn't actually passed into the listener?

**Review by Tratcher — APPROVED** [review]
Now I have to wonder if it works 😁

**Comment by jkotalik** [comment]
I'm going to try reverifying tomorrow; I had trouble building the aspnetcore repo on WSL2, going to try on a dedicated VM tomorrow.

**Comment by jkotalik** [comment]
There were some bugs to say the least. Thanks for keeping me in check @Tratcher

**Comment by jkotalik** [comment]
I'll handle updating shared code tomorrow.

**Pull request merged**